Classes/Program/index.js

```diff
@@ -0,0 +1,206 @@
+const logger = require('node-color-log');
+
+const BSTFactory = require('../BSTFactory');
+const {FileSystemReader} = require('../FileSystemJSON');
+const SortAndFind = require('../SortAndFind');
+
+/**
+ * Class for create Program
+ */
+class Program {
+  /**
+   * @param {String} pathToFile - path to file from which read data
+   * @param {SortAndFind} sortAndFind
+  */
+  constructor(pathToFile, sortAndFind) {
+    this.data = [];
+    this.onLine = this.onLine.bind(this);
+    this.sortAndFind = sortAndFind;
+    this.options = {
+      show: this.showOption.bind(this),
+      find: this.findOption.bind(this),
+      sort: this.sortedOption.bind(this),
+      help: this.logOptions.bind(this),
+    };
+    this.pathToFile = pathToFile;
+  }
+  /**
+   * @return {Promise<[]>}
+   */
+  async getData() {
+    return await FileSystemReader.read(this.pathToFile);
+  }
+  /**
+   * Log options
+   */
+  logOptions() {
+    logger.color('cyan').log('Options:');
+    logger.color('cyan').log(`
+option=help - show all operations
+
+option=show - show first 100 records
+option=show limit=1 - show first record
+option=show all=true - show all records
+option=show page=2 - show next 100 records
+
+option=find selector=s value=v - find first 100 records with value=v
+option=find selector=s value=v limit=1 - find first record with value=v
+option=find selector=s value=v all=true - find all records with value=v
+option=find selector=s value=v page=2 - find next 100 records with value=v
+
+option=sort selector=s - show first 100 records sorted by selector
+option=sort selector=s limit=1 - show first record in sorted array
+option=sort selector=s all=true - show all records sorted by selector
+option=sort selector=s order=1 - show sorted array in increase order
+option=sort selector=s order=-1 - show sorted array in decrease order
+option=sort selector=s page=2 - show next 100 sorted records
+        `);
+  }
+  /**
+   * Log error
+   * @param {String} errorText - error text
+   */
+  logError(errorText) {
+    logger.color('red').log(errorText);
+    logger
+        .color('cyan')
+        .log('Write: option=help for get more info about program');
+  }
+  /**
+   * Main function
+   */
+  async main() {
+    this.data = await this.getData();
+    this.onLine();
+  }
+  /**
+   * Event listener for readLine
+   */
+  onLine() {
+    const option = this.getOptionFromInput();
+    this.callOption(option);
+  }
+  /**
+   * @return {String}
+   */
+  getOptionFromInput() {
+    const option = process.env.option;
+    return option;
+  }
+
+  /**
+   * @param {String} option - option
+   * @return {void}
+   */
+  callOption(option) {
+    if (!this.options[option]) {
+      this.logError(`No such option like: '${option}'`);
```

Comment:
no need to use the separate method for logging error, see comment about logger class